modules/gapi/src/api/gcomputation.cpp

```diff
@@ -165,6 +165,42 @@ void cv::GComputation::apply(const std::vector<cv::Mat> &ins,
     apply(std::move(call_ins), std::move(call_outs), std::move(args));
 }
 
+cv::GRunArgs cv::GComputation::apply(GRunArgs &&ins, GCompileArgs &&args)
+{
+    auto& shape = m_priv->m_shape;
+    GAPI_Assert(util::holds_alternative<cv::GComputation::Priv::Expr>(shape));
+    const auto& expr = cv::util::get<cv::GComputation::Priv::Expr>(shape);
+
+    GRunArgs run_args;
+    GRunArgsP outs;
+    run_args.reserve(expr.m_outs.size());
+    outs.reserve(expr.m_outs.size());
+
+    for (auto&& p : expr.m_outs)
+    {
+        switch (p.index())
+        {
+            case GProtoArg::index_of<cv::GMat>():
+            {
+                run_args.emplace_back(cv::Mat{});
+                outs.emplace_back(&cv::util::get<cv::Mat>(run_args.back()));
+                break;
+            }
+            case GProtoArg::index_of<cv::GScalar>():
+            {
+                run_args.emplace_back(cv::Scalar{});
+                outs.emplace_back(&cv::util::get<cv::Scalar>(run_args.back()));
+                break;
+            }
+            default:
+                util::throw_error(std::logic_error("Only cv::GMat and cv::GScalar are supported for input"));
```

Comment:
...for Python

Reply:
for input -> for output? You're iterating over `m_outs` here right?

Reply:
You are right `for output`